src/components/application_manager/src/commands/hmi/get_urls.cc

```diff
@@ -194,26 +194,30 @@ void GetUrls::ProcessPolicyServiceURLs(const policy::EndpointUrls& endpoints) {
 
   // Will use only one URL for particular application if it will be found
   // Otherwise URL from default section will used
-  SmartObject service_info = SmartObject(SmartType_Map);
+  SmartObject service_infos = SmartObject(SmartType_Array);
 
+  size_t count_of_services = 0;
   for (size_t e = 0; e < endpoints.size(); ++e) {
+    SmartObject service_info = SmartObject(SmartType_Map);
+
     if (mobile_app_id == endpoints[e].app_id) {
       if (endpoints[e].url.size()) {
-        service_info[url] = endpoints[e].url[0];
-        SendResponseToHMI(Common_Result::SUCCESS);
-        return;
+        for (size_t it_url = 0; it_url != endpoints[e].url.size(); ++it_url) {
```

Comment:
@VVeremjova is is possible to extract to separate functions some parts of this code.